config-inject/README.md

```diff
@@ -0,0 +1,150 @@
+export type AppConfig = {
+    env: 'staging' | 'production' | 'dev'
+    version: string
+    nested: {
+        boi: string
+    }
+}
+```
+
+Then in your per-env config files use use TypeScript via JSDocs annotations, e.g.:
+
+```mjs
+// @ts-check
+/**
+ * @type {import("./config").AppConfig}
+ */
+export const config = {
+    env: 'staging',
+    version: '__INJECTED__',
+    nested: {
+        boi: 'wat'
+    }
+}
+```
+
+2. Consume the config in your app through a module exposing the typed config:
+
+```ts
+// config/index.ts
+import {getConfig} from '@pleo-io/spa-config-inject'
+import type {AppConfig} from './config'
+
+export default getConfig<AppConfig>()
+```
+
+Then in your app code:
+
+```ts
+// some-file.ts
+import config from './config'
+
+let boi = config.nested.boi // string
+```
+
+## Configuration
+
+You can configure the tool by adding a spaConfig namespace to your package.json file. The
+configuration options are as follows:
+
+-   `configDir` (string): The location of your app configuration files.
+-   `buildDir` (string): The location of the HTML template file.
+-   `templateFileName` (string, optional, default: `_index.html`): The name of the template HTML
+    file (relative to build dir).
+-   `outputFileName` (string, optional, default: `index.html`): The name of the output HTML file
+    (relative to build dir).
+-   `devConfigOverrideFile` (string, optional, default: `config.dev.json`): The name of the
+    development config override file.
+
+## Vite plugin
+
+To inject the dev config when running the app locally, add the Vite plugin to your Vite config:
+
+```ts
+import {defineConfig} from 'vite'
+import {inlineDevelopmentConfig} from '@pleo-io/spa-config-inject/vite'
+
+export default defineConfig((config) => {
+    return {
+        plugins: [inlineDevelopmentConfig(config)]
+    }
+})
+```
+
+## Dev overrides
+
+When running the app in dev mode the first time, an empty `config.dev.json` (or another name if you
+configure it via `devConfigOverrideFile` option) will be created at the root of the project. This
+file allows you to temporarily override the dev config (e.g. change an API route) without having to
+remember to undo your changes before committing. Remember to add that file to `.gitingored` as it's
+not supposed to be versioned.
+
+## CLI Usage
+
+In your CD flow you'll need to run the CLI for each environment you deploy to, in order to generate
+the HTML with baked in config for that env.
+
+```sh
+$ pnpm spa-config-inject --env <env>
+```
+
+Usually you'll also want to add some dynamic overrides based on your CD run, e.g. adding a version:
+
+```sh
+$ SPA_CONFIG_OVERRIDE="{\"version\": \"${{github.sha}}\"}" pnpm spa-config-inject --env staging
+```
+
+### Options
+
+Required options:
+
+-   `--env`: The environment of the config to inject. Can also be provided via `SPA_ENV` env var.
+
+Optional options:
+
+-   `--dynamic-config`: Dynamic config override as a JSON string. Can also be provided via
```

Comment:
So we have 3 different API surfaces: package.json, `SPA_` environment variables and CLI options. Wouldn't it make sense to only support one of them? I find the package.json interface neat and suggest getting rid of the others, expect for `SPA_CONFIG_OVERRIDE` and `SPA_ENV`.

If we do that then I we can also get rid of commander, which is a fairly big dependency.